### ISSUE TITLE
TChain doesn't recognise ROOT::VecOps::RVec<double> as a column type

### ISSUE THREAD

**Issue by guar1337** [root-project/root#10225]
- [x] Checked for duplicates

### Describe the bug
TChain doesn't recognise ROOT::VecOps::RVec<double> as a column type. "Error in <TBranch::TLeaf>: Illegal data type for ..."

### Expected behavior
I would expect TChain to recognize ROOT::VecOps::RVec<double>.

### To Reproduce
```

bool flag = false;
struct vecMaker
{
	std::vector<Double_t> operator()()
    {
        std::vector<Double_t> myVec{0,1,2,3,4,5};
        return myVec;
    }
};


ROOT::RDF::RNode ApplyDefines(ROOT::RDF::RNode df)
{
	if (flag)
	{
		return df.Define("y", [](){ return gRandom->Rndm(); });
	}
    flag = true;
	return ApplyDefines(df.Define("RVecColumn", vecMaker()));
}

void clipboard()
{
    ROOT::EnableImplicitMT();
    //first part of the 
    auto unifGen = [](double) { return gRandom->Uniform(-1.0, 1.0); };
    auto vGen = [&](int len)
    {
        ROOT::VecOps::RVec<Double_t> v(len);
        std::transform(v.begin(), v.end(), v.begin(), unifGen);
        return v;
    };

    ROOT::RDataFrame d(1024);
    auto outDF = d.Define("len", []() { return (int)gRandom->Uniform(0, 16); })
      .Define("x", vGen, {"len"});

    //ApplyDefines forces RDataFrame to save a column as a ROOT::VecOps::RVec<double>
	auto myNewDF = ApplyDefines(outDF);
    myNewDF.Snapshot("myDF", "mydf.root");

	TChain dEChain("myDF");
	dEChain.Add("mydf.root");

	ROOT::RDataFrame dEDF(dEChain);
	dEDF.Snapshot("analyzed", "newdf.root");
}
```

### Setup
1. ROOT version: 6.26/00
2. Operating system: Ubuntu 20.04
3. How you obtained ROOT, such as `dnf install` / binary download / you built it yourself: built from source

### Additional context
In the reproducer code, I forced RDataFrame to save a column as a "ROOT::VecOps::RVec<double>". This is a solution I am actually using in my code: https://root-forum.cern.ch/t/passing-arguments-to-function-called-by-define-in-rdataframe/42208/3

**Comment by eguiraud** [comment]
Hi @guar1337 ,
thank you for your report, I will take a look as soon as possible.

**Comment by eguiraud** [comment]
@guar1337 can you confirm that this reproduces the issue when run as `root -l -b -q repro.C`?

```cpp
void repro() {
  {
    ROOT::RDataFrame d(1);
    auto outDF = d.Define("x", [] { return ROOT::RVecD{1., 2.}; });
    outDF.Snapshot<ROOT::RVecD>("myDF", "mydf.root", {"x"});
  }

  {
    ROOT::RDataFrame dEDF("myDF", "mydf.root");
    dEDF.Snapshot<ROOT::RVecD>("analyzed", "newdf.root", {"x"});
  }
}
```

**Comment by guar1337** [comment]
Yes, still the same problem. newdf.root is being created, but it's empty. I am 99% sure, that I didn't have this issue with root 6.24.02. The behavior changed after upgrading, hopefully, I didn't mess anything up during installation.

**Comment by eguiraud** [comment]
No no, this is a bug that slipped our tests: it requires reading RVecs from file and writing them out again, while we only test either reading them from file or writing them from memory.

I'll fix this for 6.26.02, which is coming out this or next week.

**Comment by eguiraud** [comment]
Thank you again for reporting the problem @guar1337 , this is fixed in master and the fix will be included in the upcoming v6.26.02 patch release.

**Comment by guar1337** [comment]
You guys are awesome, thanks!
